py/DREAM/Settings/Equations/IonSpecies.py

```diff
@@ -447,6 +614,187 @@ def initialize_prescribed_charge_state(self, Z0, n=None, r=None, t=None, interpr
             self.initialize_prescribed(n=N, t=t, r=r)
         else:
             raise EquationException("ion_species: '{}': Unrecognized shape of prescribed density: {}.".format(self.name, n.shape))
+            
+            
+            
+            
+    def initialize_charged_prescribed_diffusion(self, charged_prescribed_diffusion=None, rChargedPrescribedDiffusion=None, tChargedPrescribedDiffusion=None, interpr=None, interpt=None):
```

Comment:
General comment about all these ``initialize`` routines: do you think it would be possible to put most of the logic in a general initialization routine which utilizes Python's dynamic variables to avoid duplicating code? I imagine that these initialization methods would remain, but their contents would be reduced to calling a general method which also takes an additional parameter which indicates the name of the relevant variables to set. Instead of setting ``self.tChargedPrescribedDiffusion`` explicitly, one would then call ``setattr(self, f"t{name}PrescribedDiffusion", value)``, or something like that.

(This is how it's done in the general transport interface, ``py/DREAM/Settings/Transport.py``)